### PR TITLE
add example for echo string interpolation and evaluation

### PR DESCRIPTION
**Pull request by jordanst3wart**
# Description
Add example for echo string interpolation and evaluation

# User-Facing Changes
Adds an example in the document for [the echo command](https://www.nushell.sh/commands/docs/echo.html)

### CODE REVIEW

File: crates/nu-cmd-lang/src/core_commands/echo.rs [hunk 1]
```diff
@@ -58,6 +58,14 @@ little reason to use this over just writing the values as-is."#
                 example: "echo $in",
                 result: None,
             },
+            Example {
+                description: "To do string interpolation, and evaluation inside a string.",
+                example: r#""Alice" | echo $"greetings, ($in)" $"(2 + 2)""#,
```
Comment (fdncred) [inline crates/nu-cmd-lang/src/core_commands/echo.rs:63]:
```suggestion
                example: r#""Alice" | echo $'greetings, ($in) (2 + 2)'#,
```

File: crates/nu-cmd-lang/src/core_commands/echo.rs [hunk 2]
```diff
@@ -58,6 +58,14 @@ little reason to use this over just writing the values as-is."#
...
+                result: Some(Value::list(
+                    vec![Value::test_int(1), Value::test_int(2)],
```
Comment (fdncred) [inline crates/nu-cmd-lang/src/core_commands/echo.rs:65]: This result is wrong for the example above.

**Comment by jordanst3wart** [comment]
This relates heavily to string here: https://www.nushell.sh/book/working_with_strings.html#string-interpolation

I'm guessing these samples should still help as `echo` is used a lot in other shells

**Comment by fdncred** [comment]
The problem with your examples is that `echo` is not required.

**Comment by jordanst3wart** [comment]
I could change it to something like:
`'Alice' | echo $'Name: ($in)'`
`2 | echo $'2 + ($in) = (2 + $in)'`
I think that errors the example tests though

**Comment by fdncred** [comment]
I guess my point is that even those examples do not require `echo`. Both of these work without echo.
```nushell
'Alice' | $'Name: ($in)'
2 | $'2 + ($in) = (2 + $in)'
```

**Comment by jordanst3wart** [comment]
In this example you need `echo`:
```
❯ 'Alice' | echo $'Name: ($in)' $'2 + 2 = (2 + 2)'
╭───┬─────────────╮
│ 0 │ Name: Alice │
│ 1 │ 2 + 2 = 4   │
╰───┴─────────────╯
```
Although, it's probably better to just use a list

**Comment by fdncred** [comment]
ya, at least that one needs echo.

**Comment by rgwood** [comment]
Thanks for the PR!

> Although, it's probably better to just use a list

Agreed.

`echo` is a bit of a weird command, it's mostly in Nu for compatibility with other shells and usually isn't needed. I think I would vote to not add these examples, I don't want users to think that they _should_ be using `echo` this way.

**Comment by fdncred** [comment]
Thanks for trying to make nushell easier to understand. I think we should pass on this one too.

**Pull request closed without merging**